src/components/QueryEditor.tsx

```diff
@@ -69,7 +69,19 @@ export function QueryEditor(props: Props) {
   };
 
   const handleAggregationChange = (item: SelectableValue) => {
-    props.onChange({ ...query, metric: item.value });
+    // set a default value when percentile is selected and delete percentileValue when percentile is deselected
+    // to not pollute the dashboard.json file
+    let percentile = undefined;
+    if (item.value === 'percentile' && percentileValue == null) {
+      setPercentileValue(95);
+      percentile = 95;
+    } else {
+      if (percentileValue != null) {
+        setPercentileValue(undefined);
+      }
+    }
```

Comment:
why not simple `else` , why is there another `if` ? 
```suggestion
    } else {
        setPercentileValue(undefined);
    }
```

Reply:
My thinking was that if the percentileValue is already null we do not need to update the state becuase it already would be null/undefined. 🤔